### ISSUE TITLE
There should be an some sort of option to mark commands as dangerous

### ISSUE THREAD

**Issue by StayBlue** [Epix-Incorporated/Adonis#1350]
### What part of Adonis is this related to?

Command

### What are you suggesting?

Currently, some commands use a Yes or No prompt before executing. Currently, it is possible to `:ban all` as an Admin, and there will be no warning prompt suggesting that this might possibly be a typo. There should be some way to mark commands as dangerous, causing a prompt before executing commands if a certain number of players are affected.

This could be retroactively applied to existing commands where it is deemed necessary. If this is done, making it a settings option in the Loader might make sense.

**Comment by Dimenpsyonal** [comment]
So something like `settings.WarnDangerousActions`?

**Comment by StayBlue** [comment]
> So something like `settings.WarnDangerousActions`?

That sounds like a good name for it. I'm currently working on this issue myself. The idea I came up with is if the setting is enabled, it will check if arguments are wrapped with certain symbols:
```lua
server.Commands.Danger = {
	-- Rest of the code   
	Args = {"!players!"},
}
```
In this example, exclamation marks indicate that it should expect that:
1) The argument is a list of players
2) The argument is potentially dangerous

I haven't decided how I'll check if it's dangerous or not. I'm still unfamiliar with the Adonis codebase and trying to familiarize myself with it, so if you have any suggestions, please let me know!

**Comment by Dimenpsyonal** [comment]
Alternatively you could add an argument in the top like Fun or Disabled, like so

```lua
server.Commands.Ban= {
	Dangerous = true;
	Args = {"player(s)", "reason"},
}```

And then if the `all` selector is used then the yes/no will appear

**Comment by StayBlue** [comment]
My question, then, is how would I check which argument is expected to be a list of usernames?

To demonstrate the current dilemma, the `Kick` command lists `player` as the first argument, the `Ban` command lists `player/user` as the first argument, and the `DirectTimeBan` command lists `username(s)` as the first argument. All of these except a list of usernames, but say so differently.

**Comment by StayBlue** [comment]
Now that you mention it, I find it strange that you still get auto-complete suggestions for players even when the argument isn't expecting a player. It might be worth creating a system for noting what's expected within a given argument, but the only worry there is backward compatibility.

**Comment by ccuser44** [comment]
> Now that you mention it, I find it strange that you still get auto-complete suggestions for players even when the argument isn't expecting a player. It might be worth creating a system for noting what's expected within a given argument, but the only worry there is backward compatibility.

I think the system should just check if the arg name is "`player`" or "`plr`"

Having a more complex system to validate if an arg is a player is probably a bad idea because of bloat

**Comment by ccuser44** [comment]
All `user` and `player/user` should just be converted to `player` even if it refers to offline players

**Comment by ccuser44** [comment]
> > All `user` and `player/user` should just be converted to `player` even if it refers to offline players
> 
> Maybe this is a little pessimistic, but what if a command expects two separate lists of players?

`player` and `player` ¯\_(ツ)_/¯

**Comment by StayBlue** [comment]
> > > All `user` and `player/user` should just be converted to `player` even if it refers to offline players
> > 
> > 
> > Maybe this is a little pessimistic, but what if a command expects two separate lists of players?
> 
> `player` and `player` ¯\_(ツ)_/¯

Fair enough. I was only thinking about it from a UX perspective, given that there would be no way to tell the user what each player list means with this idea.

**Comment by ccuser44** [comment]
> > > > All `user` and `player/user` should just be converted to `player` even if it refers to offline players
> > > 
> > > 
> > > Maybe this is a little pessimistic, but what if a command expects two separate lists of players?
> > 
> > 
> > `player` and `player` ¯_(ツ)_/¯
> 
> Fair enough. I was only thinking about it from a UX perspective, given that there would be no way to tell the user what each player list means with this idea.

I see
I'm not sure what exactly should be done

Anyways the `user` thing or `player/user` should just be replaced with `player`

**Comment by Dimenpsyonal** [comment]
Yes
